Tools/CISupport/build-webkit-org/steps.py

```diff
@@ -205,6 +206,39 @@ class KillOldProcesses(shell.Compile):
     command = ["python3", "Tools/CISupport/kill-old-processes", "buildbot"]
 
 
+class PruneCoreSymbolicationdCacheIfTooLarge(buildstep.BuildStep):
+    name = "prune-coresymbolicationd-cache-if-too-large"
+    description = ["pruning coresymbolicationd cache to < 10GB"]
+    descriptionDone = ["pruned coresymbolicationd cache"]
+    flunkOnFailure = False
+    haltOnFailure = False
+
+    def __init__(self):
+        buildstep.BuildStep.__init__(self)
+
+    def start(self):
+        directory = '/System/Library/Caches/com.apple.coresymbolicationd'
+        directoryPath = Path(directory).resolve()
+
+        if not directoryPath.exists():
+            self._addToLog('stdio', 'coresymbolicationd cache does not exist')
+            self.finished(SKIPPED)
+            return None
+
+        if not directoryPath.is_dir():
+            self._addToLog('stdio', 'coresymbolicationd cache is not a directory')
+            self.finished(FAILURE)
+            return None
+
+        tenGigabytes = 1024 * 1000 * 1000 * 10
```

Comment:
Nit: I guess you meant 1024 instead of 1000

Reply:
Fixed.